lib/ecto/repo/schema.ex

```diff
@@ -467,7 +467,7 @@ defmodule Ecto.Repo.Schema do
           case apply(user_changeset, adapter, action, args) do
             {:ok, values} ->
               changeset
-              |> load_changes(:loaded, return_types, values, embeds, autogen, adapter, schema_meta)
+              |> load_changes(:loaded, return_types, values, assocs, embeds, autogen, adapter, schema_meta)
```

Comment:
We have `parents`, which already returns only the parents assocs, reducing the amount we need to traverse. Although I wonder if we could move the cleaning up to `process_parents`?

Reply:
Right now only the changed parents are kept which are being taken care of here: https://github.com/elixir-ecto/ecto/blob/master/lib/ecto/repo/schema.ex#L925.

But if we start keeping the unchanged parents we can do better. What do you think of the latest commit?